core/src/main/java/org/elasticsearch/action/admin/indices/rollover/RolloverResponse.java

```diff
@@ -42,134 +46,212 @@
+                String condition = in.readString();
+                boolean satisfied = in.readBoolean();
+                conditions.add(new AbstractMap.SimpleEntry<>(condition, satisfied));
+            }
+            response.conditionStatus = conditions;
+            response.dryRun = in.readBoolean();
+            response.rolledOver = in.readBoolean();
+            response.acknowledged = in.readBoolean();
+            response.shardsAcked = in.readBoolean();
+            responses.add(response);
+        }
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
-        out.writeString(oldIndex);
-        out.writeString(newIndex);
-        out.writeVInt(conditionStatus.size());
-        for (Map.Entry<String, Boolean> entry : conditionStatus) {
-            out.writeString(entry.getKey());
-            out.writeBoolean(entry.getValue());
-        }
-        out.writeBoolean(dryRun);
-        out.writeBoolean(rolledOver);
-        out.writeBoolean(acknowledged);
-        out.writeBoolean(shardsAcked);
+        out.writeVInt(responses.size());
+        for (SingleAliasRolloverResponse response: responses) {
+            out.writeString(response.oldIndex);
+            out.writeString(response.newIndex);
+            out.writeVInt(response.conditionStatus.size());
+            for (Map.Entry<String, Boolean> entry : response.conditionStatus) {
+                out.writeString(entry.getKey());
+                out.writeBoolean(entry.getValue());
+            }
+            out.writeBoolean(response.dryRun);
+            out.writeBoolean(response.rolledOver);
+            out.writeBoolean(response.acknowledged);
+            out.writeBoolean(response.shardsAcked);
+        }
     }
 
     @Override
-    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+    public XContentBuilder toXContent(final XContentBuilder builder, Params params) throws IOException {
+        if (responses.isEmpty() && failures.isEmpty()) {
+            throw new IllegalStateException("Response for at least 1 alias is expected");
+        }
+
         builder.startObject();
-        builder.field(OLD_INDEX, oldIndex);
-        builder.field(NEW_INDEX, newIndex);
-        builder.field(ROLLED_OVER, rolledOver);
-        builder.field(DRY_RUN, dryRun);
-        builder.field(ACKNOWLEDGED, acknowledged);
-        builder.field(SHARDS_ACKED, shardsAcked);
-        builder.startObject(CONDITIONS);
-        for (Map.Entry<String, Boolean> entry : conditionStatus) {
-            builder.field(entry.getKey(), entry.getValue());
+        builder.field(DRY_RUN, responses.get(0).isDryRun());
+
+        // keep compatibility with single alias Rollover
+        if (responses.size() == 1) {
+            first().toXContent(builder, params);
+        }
+
+        builder.startObject("rolled_over_aliases");
+        for(SingleAliasRolloverResponse response: this.responses) {
```

Comment:
Yes, it would, and it made intentionally. In case there is only one response we return this response
1) inline (fully compatible with current API)
2) as element of "rolled_over_aliases"
We discussed it with @s1monw in #25506

Reply:
Okay, I see. I understood his comment more like an either/or, so with single alias inline, with multiple use a new object but don't inline the first one. Let's see what he thinks about it.